### ISSUE TITLE
"Proprietary" license or GPL-2.0?

### ISSUE THREAD

**Issue by sneakers-the-rat** [nel-lab/FIOLA#32]
Hello!

i'd like to try this software, but there is conflicting package metadata re: the license - 

There is a copy of the GPL-2.0 in `LICENSE`:
https://github.com/nel-lab/FIOLA/blob/ff41b86f0d3e494086f44bf01bc8b6739e1aae0a/LICENSE 

but the python package metadata indicates that the license is "Proprietary":
https://github.com/nel-lab/FIOLA/blob/ff41b86f0d3e494086f44bf01bc8b6739e1aae0a/setup.py#L19

With a "Proprietary" license (or without any license that has an explicit grant of permissions to use and redistribute it) the default is that it is [illegal without permission from the author to run the software](https://www.synopsys.com/blogs/software-security/unlicensed-open-source-scenarios.html). Naturally this is pretty bad for the health of the scientific software community, so i would suggest the authors choose the GPL over "Proprietary" --- I'm presuming the authors want people to use their tool and that was unintentional.

**Comment by caichangjia** [comment]
Sorry for the inconvenience. It is under GPL-2.0. I have updated the setup.py file. Thanks for pointing it out.

**Comment by sneakers-the-rat** [comment]
Awesome, thanks for the clarification.

I truly don't mean to be a pain (seriously!), but just want to give you a heads up - it would be extremely unlikely that anyone would try and sue you over it, but GPL-2.0 is incompatible with most contemporary licenses (GPL-3.0 is incompatible in the case of pinning version which FSF defines as creating a derivative work, Apache-2.0 is explicitly incompatible, etc.) - you have your own reasons for using it, but `GPL-2.0-or-later` is another option.

The output of `pip-licenses` as an indicator of which might be a problem, looks like it's mostly apache.

| Name                          | Version   | License                                                                                          |
|-------------------------------|-----------|--------------------------------------------------------------------------------------------------|
| asttokens                     | 2.2.1     | Apache 2.0                                                                                       |
| tensorboard-plugin-wit        | 1.8.1     | Apache 2.0                                                                                       |
| Cython                        | 0.29.33   | Apache Software License                                                                          |
| absl-py                       | 0.15.0    | Apache Software License                                                                          |
| arrow                         | 1.2.3     | Apache Software License                                                                          |
| bleach                        | 6.0.0     | Apache Software License                                                                          |
| coverage                      | 7.2.2     | Apache Software License                                                                          |
| dm-tree                       | 0.1.5     | Apache Software License                                                                          |
| flatbuffers                   | 1.12      | Apache Software License                                                                          |
| google-auth                   | 2.17.0    | Apache Software License                                                                          |
| google-auth-oauthlib          | 0.4.6     | Apache Software License                                                                          |
| google-pasta                  | 0.2.0     | Apache Software License                                                                          |
| grpcio                        | 1.32.0    | Apache Software License                                                                          |
| importlib-metadata            | 6.1.0     | Apache Software License                                                                          |
| importlib-resources           | 5.12.0    | Apache Software License                                                                          |
| keras                         | 2.12.0    | Apache Software License                                                                          |
| opencv-python                 | 4.7.0.72  | Apache Software License                                                                          |
| prometheus-client             | 0.16.0    | Apache Software License                                                                          |
| requests                      | 2.28.2    | Apache Software License                                                                          |
| rsa                           | 4.9       | Apache Software License                                                                          |
| tensorboard                   | 2.11.2    | Apache Software License                                                                          |
| tensorboard-data-server       | 0.6.1     | Apache Software License                                                                          |
| tensorflow                    | 2.4.1     | Apache Software License                                                                          |
| tensorflow-addons             | 0.12.1    | Apache Software License                                                                          |
| tensorflow-estimator          | 2.4.0     | Apache Software License                                                                          |
| tensorflow-probability        | 0.12.1    | Apache Software License                                                                          |
| tk                            | 0.1.0     | Apache Software License                                                                          |
| tornado                       | 6.2       | Apache Software License                                                                          |
| websocket-client              | 1.5.1     | Apache Software License                                                                          |
| yapf                          | 0.32.0    | Apache Software License                                                                          |
| packaging                     | 23.0      | Apache Software License; BSD License                                                             |
| python-dateutil               | 2.8.2     | Apache Software License; BSD License                                                             |
| sniffio                       | 1.3.0     | Apache Software License; MIT License                                                             |
| jupyterlab-pygments           | 0.2.2     | BSD                                                                                              |
| numpy                         | 1.19.5    | BSD                                                                                              |
| slicerator                    | 1.1.0     | BSD                                                                                              |
| matplotlib-inline             | 0.1.6     | BSD 3-Clause                                                                                     |
| Babel                         | 2.12.1    | BSD License                                                                                      |
| Jinja2                        | 3.0.1     | BSD License                                                                                      |
| Markdown                      | 3.4.3     | BSD License                                                                                      |
| MarkupSafe                    | 2.1.2     | BSD License                                                                                      |
| Pygments                      | 2.14.0    | BSD License                                                                                      |
| Send2Trash                    | 1.8.0     | BSD License                                                                                      |
| Sphinx                        | 6.1.3     | BSD License                                                                                      |
| Werkzeug                      | 2.2.3     | BSD License                                                                                      |
| alabaster                     | 0.7.13    | BSD License                                                                                      |
| appnope                       | 0.1.3     | BSD License                                                                                      |
| astunparse                    | 1.6.3     | BSD License                                                                                      |
| backcall                      | 0.2.0     | BSD License                                                                                      |
| cloudpickle                   | 3.0.0     | BSD License                                                                                      |
| comm                          | 0.1.3     | BSD License                                                                                      |
| contourpy                     | 1.0.7     | BSD License                                                                                      |
| cycler                        | 0.11.0    | BSD License                                                                                      |
| decorator                     | 5.1.1     | BSD License                                                                                      |
| fastjsonschema                | 2.16.3    | BSD License                                                                                      |
| gast                          | 0.3.3     | BSD License                                                                                      |
| h5py                          | 2.10.0    | BSD License                                                                                      |
| hdmf                          | 3.5.2     | BSD License                                                                                      |
| holoviews                     | 1.14.9    | BSD License                                                                                      |
| idna                          | 3.4       | BSD License                                                                                      |
| imageio                       | 2.27.0    | BSD License                                                                                      |
| imageio-ffmpeg                | 0.4.8     | BSD License                                                                                      |
| ipykernel                     | 6.22.0    | BSD License                                                                                      |
| ipyparallel                   | 8.5.1     | BSD License                                                                                      |
| ipython                       | 8.11.0    | BSD License                                                                                      |
| ipython-genutils              | 0.2.0     | BSD License                                                                                      |
| ipywidgets                    | 8.0.6     | BSD License                                                                                      |
| joblib                        | 1.2.0     | BSD License                                                                                      |
| jsonpointer                   | 2.3       | BSD License                                                                                      |
| jupyter                       | 1.0.0     | BSD License                                                                                      |
| jupyter-console               | 6.6.3     | BSD License                                                                                      |
| jupyter-events                | 0.6.3     | BSD License                                                                                      |
| jupyter_client                | 8.1.0     | BSD License                                                                                      |
| jupyter_core                  | 5.3.0     | BSD License                                                                                      |
| jupyter_server                | 2.5.0     | BSD License                                                                                      |
| jupyter_server_terminals      | 0.4.4     | BSD License                                                                                      |
| jupyterlab-widgets            | 3.0.7     | BSD License                                                                                      |
| kiwisolver                    | 1.4.4     | BSD License                                                                                      |
| mistune                       | 2.0.5     | BSD License                                                                                      |
| nbclassic                     | 0.5.3     | BSD License                                                                                      |
| nbclient                      | 0.7.2     | BSD License                                                                                      |
| nbconvert                     | 7.2.10    | BSD License                                                                                      |
| nbformat                      | 5.8.0     | BSD License                                                                                      |
| nest-asyncio                  | 1.5.6     | BSD License                                                                                      |
| networkx                      | 3.0       | BSD License                                                                                      |
| notebook                      | 6.5.3     | BSD License                                                                                      |
| notebook_shim                 | 0.2.2     | BSD License                                                                                      |
| numpydoc                      | 1.5.0     | BSD License                                                                                      |
| oauthlib                      | 3.2.2     | BSD License                                                                                      |
| pandas                        | 1.4.4     | BSD License                                                                                      |
| pandocfilters                 | 1.5.0     | BSD License                                                                                      |
| panel                         | 0.12.1    | BSD License                                                                                      |
| param                         | 1.13.0    | BSD License                                                                                      |
| prompt-toolkit                | 3.0.38    | BSD License                                                                                      |
| psutil                        | 5.9.4     | BSD License                                                                                      |
| pyasn1                        | 0.4.8     | BSD License                                                                                      |
| pyasn1-modules                | 0.2.8     | BSD License                                                                                      |
| pycparser                     | 2.21      | BSD License                                                                                      |
| pyct                          | 0.5.0     | BSD License                                                                                      |
| pynwb                         | 2.3.1     | BSD License                                                                                      |
| python-json-logger            | 2.0.7     | BSD License                                                                                      |
| pyviz-comms                   | 2.2.1     | BSD License                                                                                      |
| qtconsole                     | 5.4.1     | BSD License                                                                                      |
| requests-oauthlib             | 1.3.1     | BSD License                                                                                      |
| scikit-image                  | 0.19.3    | BSD License                                                                                      |
| scikit-learn                  | 1.2.2     | BSD License                                                                                      |
| scipy                         | 1.10.1    | BSD License                                                                                      |
| snowballstemmer               | 2.2.0     | BSD License                                                                                      |
| sphinxcontrib-applehelp       | 1.0.4     | BSD License                                                                                      |
| sphinxcontrib-devhelp         | 1.0.2     | BSD License                                                                                      |
| sphinxcontrib-htmlhelp        | 2.0.1     | BSD License                                                                                      |
| sphinxcontrib-jsmath          | 1.0.1     | BSD License                                                                                      |
| sphinxcontrib-qthelp          | 1.0.3     | BSD License                                                                                      |
| sphinxcontrib-serializinghtml | 1.1.5     | BSD License                                                                                      |
| terminado                     | 0.17.1    | BSD License                                                                                      |
| threadpoolctl                 | 3.1.0     | BSD License                                                                                      |
| tifffile                      | 2023.3.21 | BSD License                                                                                      |
| tinycss2                      | 1.2.1     | BSD License                                                                                      |
| traitlets                     | 5.9.0     | BSD License                                                                                      |
| webcolors                     | 1.13      | BSD License                                                                                      |
| webencodings                  | 0.5.1     | BSD License                                                                                      |
| widgetsnbextension            | 4.0.7     | BSD License                                                                                      |
| wrapt                         | 1.12.1    | BSD License                                                                                      |
| docutils                      | 0.19      | BSD License; GNU General Public License (GPL); Public Domain; Python Software Foundation License |
| pyzmq                         | 25.0.2    | BSD License; GNU Library or Lesser General Public License (LGPL)                                 |
| protobuf                      | 3.20.3    | BSD-3-Clause                                                                                     |
| PIMS                          | 0.6.1     | BSD-3-clause                                                                                     |
| colorcet                      | 3.0.1     | CC-BY License                                                                                    |
| debugpy                       | 1.6.6     | Eclipse Public License 2.0 (EPL-2.0); MIT License                                                |
| bokeh                         | 2.3.3     | Freely Distributable                                                                             |
| Pillow                        | 9.4.0     | Historical Permission Notice and Disclaimer (HPND)                                               |
| isoduration                   | 20.11.0   | ISC License (ISCL)                                                                               |
| pexpect                       | 4.8.0     | ISC License (ISCL)                                                                               |
| ptyprocess                    | 0.7.0     | ISC License (ISCL)                                                                               |
| opt-einsum                    | 3.3.0     | MIT                                                                                              |
| Keras-Preprocessing           | 1.1.2     | MIT License                                                                                      |
| PeakUtils                     | 1.3.4     | MIT License                                                                                      |
| PyWavelets                    | 1.4.1     | MIT License                                                                                      |
| PyYAML                        | 6.0       | MIT License                                                                                      |
| QtPy                          | 2.3.1     | MIT License                                                                                      |
| anyio                         | 3.6.2     | MIT License                                                                                      |
| argon2-cffi                   | 21.3.0    | MIT License                                                                                      |
| argon2-cffi-bindings          | 21.2.0    | MIT License                                                                                      |
| attrs                         | 22.2.0    | MIT License                                                                                      |
| beautifulsoup4                | 4.12.0    | MIT License                                                                                      |
| cachetools                    | 5.3.0     | MIT License                                                                                      |
| cffi                          | 1.15.1    | MIT License                                                                                      |
| charset-normalizer            | 3.1.0     | MIT License                                                                                      |
| entrypoints                   | 0.4       | MIT License                                                                                      |
| executing                     | 1.2.0     | MIT License                                                                                      |
| fonttools                     | 4.39.3    | MIT License                                                                                      |
| future                        | 0.18.3    | MIT License                                                                                      |
| imagesize                     | 1.4.1     | MIT License                                                                                      |
| jedi                          | 0.18.2    | MIT License                                                                                      |
| jsonschema                    | 4.17.3    | MIT License                                                                                      |
| mypy                          | 0.921     | MIT License                                                                                      |
| mypy-extensions               | 0.4.4     | MIT License                                                                                      |
| parso                         | 0.8.3     | MIT License                                                                                      |
| pickleshare                   | 0.7.5     | MIT License                                                                                      |
| pkgutil_resolve_name          | 1.3.10    | MIT License                                                                                      |
| platformdirs                  | 3.2.0     | MIT License                                                                                      |
| pure-eval                     | 0.2.2     | MIT License                                                                                      |
| pyparsing                     | 3.0.9     | MIT License                                                                                      |
| pyqtgraph                     | 0.12.4    | MIT License                                                                                      |
| pyrsistent                    | 0.19.3    | MIT License                                                                                      |
| pytz                          | 2023.3    | MIT License                                                                                      |
| rfc3339-validator             | 0.1.4     | MIT License                                                                                      |
| rfc3986-validator             | 0.1.1     | MIT License                                                                                      |
| ruamel.yaml                   | 0.17.21   | MIT License                                                                                      |
| ruamel.yaml.clib              | 0.2.7     | MIT License                                                                                      |
| six                           | 1.15.0    | MIT License                                                                                      |
| soupsieve                     | 2.4       | MIT License                                                                                      |
| stack-data                    | 0.6.2     | MIT License                                                                                      |
| termcolor                     | 1.1.0     | MIT License                                                                                      |
| tomli                         | 2.0.1     | MIT License                                                                                      |
| typeguard                     | 2.13.3    | MIT License                                                                                      |
| uri-template                  | 1.2.0     | MIT License                                                                                      |
| urllib3                       | 1.26.15   | MIT License                                                                                      |
| zipp                          | 3.15.0    | MIT License                                                                                      |
| tqdm                          | 4.65.0    | MIT License; Mozilla Public License 2.0 (MPL 2.0)                                                |
| certifi                       | 2022.12.7 | Mozilla Public License 2.0 (MPL 2.0)                                                             |
| fqdn                          | 1.5.1     | Mozilla Public License 2.0 (MPL 2.0)                                                             |
| defusedxml                    | 0.7.1     | Python Software Foundation License                                                               |
| matplotlib                    | 3.6.3     | Python Software Foundation License                                                               |
| typing-extensions             | 3.7.4.3   | Python Software Foundation License                                                               |
